internal/cli/auth/register.go

```diff
@@ -20,52 +20,85 @@ import (
 	"fmt"
 	"os"
 
+	"github.com/AlecAivazis/survey/v2"
 	"github.com/mongodb/mongocli/internal/cli/require"
 	"github.com/mongodb/mongocli/internal/config"
 	"github.com/pkg/browser"
 	"github.com/spf13/cobra"
+	"go.mongodb.org/atlas/auth"
 	atlas "go.mongodb.org/atlas/mongodbatlas"
 )
 
 const accountURI = "https://account.mongodb.com/account/register?fromURI=https://account.mongodb.com/account/connect"
 const govAccountURI = "https://account.mongodbgov.com/account/register?fromURI=https://account.mongodbgov.com/account/connect"
 
+type registerSurvey struct {
+	confirm func(message string, defaultResponse bool) (response bool, err error)
+}
```

Comment:
since the point here is to mock I'd use an interface rather than using functions but not a major blocker